sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -151,3 +151,26 @@ object GroupingID {
     if (SQLConf.get.integerGroupingIdEnabled) IntegerType else LongType
   }
 }
+
+object MixedExprsWithCube {
```

Comment:
If you define extractors for the mixed case, I think we need to make them more general for extracting more complicated cases, mix of cube/rollup, mix of rollup/grouping sets, ...

Reply:
> If you define extractors for the mixed case, I think we need to make them more general for extracting more complicated cases, mix of cube/rollup, mix of rollup/grouping sets, ...

Since current code only support one cube/rollup expr, so I just support one cube/rollup expr.
Since other engine support mixed case, IMO, we should and we can support these feature and  it's compatible with the previous behavior。

I will update this later.